src/Workspaces/Core/Portable/Storage/SQLite/v1/Interop/SqlConnection.cs

```diff
@@ -216,7 +235,10 @@ private void Rollback(bool throwOnError)
             => ExecuteCommand("rollback transaction", throwOnError);
 
         public int LastInsertRowId()
-            => (int)raw.sqlite3_last_insert_rowid(_handle);
+        {
+            using var _ = _handle.Lease();
+            return (int)raw.sqlite3_last_insert_rowid(_handle.DangerousGetHandle());
```

Comment:
would it make sense to have a `LeaseAndGetHandle(out var handle)` to really ensure that callers only get the handle ift hey take the lease?

Reply:
Probably not. The call to `Lease` is only required because the static methods of `raw` have the wrong signature. The fact that this pattern stands out as strange is really by design.

Reply:
This has now been simplified.